backend/graphql/resolvers/courseResolvers.ts

```diff
@@ -1,12 +1,39 @@
+import { AuthenticationError, ExpressContext } from "apollo-server-express";
 import CourseService from "../../services/implementations/courseService";
 import {
   CreateCourseRequestDTO,
   UpdateCourseRequestDTO,
   CourseResponseDTO,
   ICourseService,
 } from "../../services/interfaces/ICourseService";
+import { getAccessToken } from "../../middlewares/auth";
+import IAuthService from "../../services/interfaces/authService";
+import AuthService from "../../services/implementations/authService";
+import nodemailerConfig from "../../nodemailer.config";
+import EmailService from "../../services/implementations/emailService";
+import UserService from "../../services/implementations/userService";
+import IEmailService from "../../services/interfaces/emailService";
+import IUserService from "../../services/interfaces/userService";
+import { Role } from "../../types";
+import { CourseVisibilityAttributes } from "../../models/course.model";
 
 const courseService: ICourseService = new CourseService();
+const userService: IUserService = new UserService();
+const emailService: IEmailService = new EmailService(nodemailerConfig);
+const authService: IAuthService = new AuthService(userService, emailService);
+
+const getCourseVisibilityAttributes = (
+  role: Role,
+): CourseVisibilityAttributes => {
+  switch (role) {
+    case "User":
+      return { private: false, published: true };
+    case "Staff":
+      return { published: true };
+    default:
```

Comment:
need to find a way to indicate that the switch case is exhaustive, sub default case with "Admin"